dawn/src/dawn/Unittest/IIRBuilder.h

```diff
@@ -91,11 +92,11 @@ class IIRBuilder {
     DAWN_ASSERT(si_);
     auto v_str = std::to_string(std::forward<T>(v));
     int acc = si_->getMetaData().insertAccessOfType(iir::FieldAccessType::FAT_Literal, v_str);
-    auto expr = std::make_shared<iir::LiteralAccessExpr>(
+    std::shared_ptr<iir::LiteralAccessExpr> expr = std::make_shared<iir::LiteralAccessExpr>(
```

Comment:
isn't `auto` here a perfect fit, because the type is completely obvious from the RHS?

Reply:
yes

Reply:
compiler interprets `auto` as `std::shared_ptr<iir::Expr>` instead of `std::shared_ptr<iir::LiteralAccessExpr>` ... therefore cannot get data

Reply:
hmm, why should it do that? `std::make_shared<X>` always returns `std::shared_ptr<X>`. If `X` is derived from `Y`, it can be cast to `std::shared_ptr<Y>`, but if you just put `auto` there is no reason why the compiler should convert it.

Reply:
So ... When compiling with gcc the error wasn't clear, it somehow suggested that it didn't find the `getData` templated method, so I thought it interpreted auto as `std::shared_ptr<iir::Expr>` ... Now I tried compiling with clang and the error says:
```
/home/giacomo/dawn/dawn/src/dawn/Unittest/IIRBuilder.h:99:11: error: use 'template' keyword to treat 'getData' as a dependent template name
    expr->getData<IIRAccessExprData>().AccessID = std::make_optional(acc);
          ^
          template
```
... now I know what to do :)